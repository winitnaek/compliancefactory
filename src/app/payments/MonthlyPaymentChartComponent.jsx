import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import JqxChart from '../../deps/jqwidgets-react/react_jqxchart.js';
import JqxButton from '../../deps/jqwidgets-react/react_jqxbuttons.js';
class MonthlyPaymentChartComponent extends Component {
    render(){
        let mothlyChartData = this.props.monthlychartdata;
        let padding = { left: 5, top: 5, right: 5, bottom: 5 };
        let titlePadding = { left: 90, top: 0, right: 0, bottom: 10 };
        let xAxis =
            {
                dataField: 'Month',
                showGridLines: true
            };
        let seriesGroups =
            [
                {
                    type: 'column',
                    columnsGapPercent: 50,
                    seriesGapPercent: 0,
                    valueAxis:
                    {
                        unitInterval: 500,
                        minValue: 0,
                        maxValue: 5000,
                        displayValueAxis: true,
                        description: 'Payment in thousands',
                        axisSize: 'auto',
                        tickMarksColor: '#888888'
                    },
                    series: [
                        { dataField: 'Keith', displayText: 'Keith' },
                        { dataField: 'Erica', displayText: 'Erica' },
                        { dataField: 'George', displayText: 'George' }
                    ]
                }
            ];
        return (
            <div className="cBox boxTools col-md-8 col-md-offset-2 col-sm-10 col-sm-offset-1"> 
            <h1>Monthly Payments</h1>  
            <JqxChart style={{ width: 850, height: 500 }} className='monthlyChart'
                title={'Monthly Payment to users'} description={'Payment received by each user in perticular month'}
                showLegend={true} enableAnimations={true} padding={padding}
                titlePadding={titlePadding} source={mothlyChartData} xAxis={xAxis}
                colorScheme={'scheme01'} seriesGroups={seriesGroups}
            />
            <JqxButton style={{ float: 'left' }} ref='printBtn' value='Print' width={80} />
            </div>
        )
    }
    componentDidMount() {
        this.refs.printBtn.on('click', () => {
            let content = window.document.getElementsByClassName('monthlyChart')[0].outerHTML;
            let newWindow = window.open('', '', 'width=800, height=500'),
                document = newWindow.document.open(),
                pageContent =
                    '<!DOCTYPE html>' +
                    '<html>' +
                    '<head>' +
                    '<meta charset="utf-8" />' +
                    '<title>Monthly Payment to users</title>' +
                    '</head>' +
                    '<body>' + content + '</body></html>';
            try {
                document.write(pageContent);
                document.close();
                newWindow.print();
                newWindow.close();
            }
            catch (error) {
            }
        });
    }
}
export default MonthlyPaymentChartComponent;