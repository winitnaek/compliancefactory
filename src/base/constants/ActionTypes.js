export const LOAD_EMPLOYEES='LOAD_EMPLOYEES'
export const LOAD_EMPLOYEES_SUCCESS='LOAD_EMPLOYEES_SUCCESS'
export const CREATE_EMPLOYEE='CREATE_EMPLOYEE'
export const CREATE_EMPLOYEE_SUCCESS='CREATE_EMPLOYEE_SUCCESS'
export const DELETE_EMPLOYEE_SUCCESS='DELETE_EMPLOYEE_SUCCESS'
export const LOAD_MONTHLYCHART_DATA='LOAD_MONTHLYCHART_DATA'
export const LOAD_MONTHLYCHART_DATA_SUCCESS='LOAD_MONTHLYCHART_DATA_SUCCESS'
export const LOAD_IMPORT_SUCCESS='LOAD_IMPORT_SUCCESS'
export const CREATE_IMPORT_SUCCESS='CREATE_IMPORT_SUCCESS'
export const DELETE_IMPORT_SUCCESS='DELETE_IMPORT_SUCCESS'
export const LOAD_APPCONF_SUCCESS='LOAD_APPCONF_SUCCESS'