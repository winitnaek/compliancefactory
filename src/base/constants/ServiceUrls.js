export const EMPLOYEES_URL='employees/'
export const MONTHLYCHART_DATA_URL='monthlyChartdata/'
export const IMPORT_URL='importdata/'

