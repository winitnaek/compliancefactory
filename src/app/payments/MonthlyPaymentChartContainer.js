//import React, {PropTypes,Component} from 'react';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import {connect} from 'react-redux';
import MonthlyPaymentChartComponent from './MonthlyPaymentChartComponent.jsx';

class MonthlyPaymentChartContainer extends Component {
    
    renderMonthlyChart(monthlychartdata){
        if(monthlychartdata && monthlychartdata.length >0){
            return(
                <MonthlyPaymentChartComponent monthlychartdata={monthlychartdata}/>
            )
        }else{
            return(<div>Loading Monthly Chart</div>);
        }
    }
    render(){
        return (
        <div className="cBox boxTools col-md-8 col-md-offset-2 col-sm-10 col-sm-offset-1">
            {this.renderMonthlyChart(this.props.monthlychartdata)}
        </div>
        )
    }
};
MonthlyPaymentChartContainer.propTypes = {
    monthlychartdata: PropTypes.array.isRequired
};
function mapStateToProps(state) {
  return {
    monthlychartdata: state.monthlychartdata
  }
}
export default connect(mapStateToProps)(MonthlyPaymentChartContainer);