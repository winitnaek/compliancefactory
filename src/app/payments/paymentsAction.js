import * as types from '../../base/constants/ActionTypes'
import paymentsApi from './paymentsAPI';

export function loadMonthlyChartDataSuccess(monthlychartdata) {
    return {type:types.LOAD_MONTHLYCHART_DATA_SUCCESS, monthlychartdata};
}
export function loadMonthlyChartData() {
    return function(dispatch,getState) {
        const state = getState();
        return paymentsApi.getMonthlyChartData(state.appconf.SVCS_CONTEXT_URL).then(monthlychartdata => {
        dispatch(loadMonthlyChartDataSuccess(monthlychartdata));
        }).catch(error => {
        throw(error);
        });
    };
}