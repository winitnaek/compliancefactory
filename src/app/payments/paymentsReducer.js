import * as types from '../../base/constants/ActionTypes';
import initialState from '../../base/config/initialState';

export default function paymentsReducer(state = initialState.monthlychartdata,action) {
  switch(action.type) {
    case types.LOAD_MONTHLYCHART_DATA_SUCCESS:
    return Object.assign([], ...state, action.monthlychartdata)  
    default: 
      return state;
  }
}