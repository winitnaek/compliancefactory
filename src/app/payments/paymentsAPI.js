import * as svcs from '../../base/constants/ServiceUrls'
class paymentsAPI {
  static getMonthlyChartData(apiUrl) {
    var svcs_url = `${apiUrl}${svcs.MONTHLYCHART_DATA_URL}`;
    return fetch(svcs_url).then(response => {
      return response.json();
    }).catch(error => {
      return error;
    });
  }
}
export default paymentsAPI;