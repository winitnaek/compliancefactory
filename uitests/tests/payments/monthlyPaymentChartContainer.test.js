import React from 'react';

import MonthlyPaymentChartContainer from '../../../src/app/payments/MonthlyPaymentChartContainer';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import TestRenderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import enzyme,{shallow} from  'enzyme'; 
import shallowWithStore from '../../utils/shallowWithStore.jsx'
import {getMonthlyChartData} from '../../../src/app/payments/paymentsAPI';
import * as svcs from '../../../../garnishment/src/base/constants/ServiceUrls';
import * as types from '../../../../garnishment/src/base/constants/ActionTypes';
import fetchMock from 'fetch-mock';

 const mockStore = configureMockStore();
 
 describe('Monthly chart ', () => {

  let store
  beforeEach(function() {
    store = mockStore({monthlychartdata:[]});
    enzyme.configure({ adapter: new Adapter() });
  }); 
  it('renders Loading chart to test empty chart data', () => {
    const rendered = renderer.create(
      <Provider store={store}>
      <MonthlyPaymentChartContainer/>
      </Provider>
    );
    expect(rendered.toJSON()).toMatchSnapshot(<div className="cBox boxTools col-md-8 col-md-offset-2 col-sm-10 col-sm-offset-1">
        <div>
            Loading Monthly Chart
        </div>
        </div>);
    });
});