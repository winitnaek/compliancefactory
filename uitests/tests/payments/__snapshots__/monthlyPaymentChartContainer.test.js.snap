// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
<div
  className="cBox boxTools col-md-8 col-md-offset-2 col-sm-10 col-sm-offset-1"
>
  <div>
    Loading Monthly Chart
  </div>
</div>
`;